modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -3895,20 +3879,22 @@ private Set<RemoteSnapshotFilesRecevier> activeTasks() {
                 else if (msg instanceof SnapshotFilesFailureMessage) {
                     SnapshotFilesFailureMessage respMsg0 = (SnapshotFilesFailureMessage)msg;
 
-                    RemoteSnapshotFilesRecevier task = active;
+                    synchronized (this) {
+                        RemoteSnapshotFilesRecevier task = active;
```

Comment:
Same, `RemoteSnapshotFilesRecevier task = active;` is redundant.

Reply:
Fixed